app/src/main/java/com/github/se/eduverse/repository/NotificationRepository.kt

```diff
@@ -0,0 +1,142 @@
+package com.github.se.eduverse.repository
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import androidx.core.app.NotificationCompat
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import androidx.work.workDataOf
+import com.github.se.eduverse.MainActivity
+import com.github.se.eduverse.model.NotifAutorizations
+import com.github.se.eduverse.model.Scheduled
+import com.github.se.eduverse.model.ScheduledType
+import java.util.concurrent.TimeUnit
+
+// Parameter authorizations is not used for now, it is here to make the class easier to upgrade
+open class NotificationRepository(
+    context: Context,
+    val autorizations: NotifAutorizations,
+    val workManager: WorkManager = WorkManager.getInstance(context)
+) {
+  /**
+   * Prepare a notification for an event or a task
+   *
+   * @param scheduled the event/task scheduled
+   */
+  open fun scheduleNotification(scheduled: Scheduled) {
```

Comment:
Taking into account the comment i made on NotificationData, you could change the parameter to be of type NotificationDestination and then retrieve it in main with intent.getBundleExtra, this will simplify the code and even make getScheduledById function added to TimeTableViewModel not needed anymore